src/server.rs

```diff
@@ -60,3 +76,31 @@ pub fn handle_connection(
         }
     }
 }
+
+pub fn new_p2p_connection(
+    conn_id: u64,
+    peer_address: String,
+    peer_port: u16,
+    messenger: Sender<MessengerOperations>,
+    receiver: Receiver<P2POperations>,
+) -> Uuid {
+    let peer_info = format!("{}:{}", peer_address, peer_port.to_string());
+    let stream = TcpStream::connect(peer_info).unwrap();
+
+    messenger
+        .send(MessengerOperations::New(NewConnectionMessage {
+            conn_id,
+            socket: stream.try_clone().unwrap(),
+        }))
+        .unwrap();
+
+    send_p2p_handshake(

```

Comment:
Same as my above comment- the server has no reason to know about any packets

Reply:
But if the server doesn't know about the peers nor the packets who is calling for the function in peer_conn_protocol.rs ?

Reply:
Yeah I guess we did say we were going to call it right from the server. I'm second guessing that now though, now that we have uuid connection ids we could just make the call from main since we just want to make sure that we connect to the peer right at the start of the program.

Regardless of where it's called though, the fact that a handshake is involved should be opaque to the caller, they should just know that they're adding a peer